BPOR/BPOR.Rms/Controllers/ResearcherController.cs

```diff
@@ -479,6 +479,7 @@ public async Task<IActionResult> Edit(int id,
 
         ValidateMandatoryFields(model);
 
+        model.Id = id;
```

Comment:
I would literally put this as the first line.

There may be future changes to `ValidateMandatoryFields()` that might expect `model.Id` to be populated.